### PR TITLE
remove incorrect gradient_accumulation_steps scaling in trainer

### PR DESCRIPTION
**Pull request by kashif**
removed incorrect `gradient_accumulation_steps` scaling in the trainer

## Summary by CodeRabbit

- **Refactor**
  - Simplified the tracking and logging of training metrics to offer clearer and more reliable performance information during training.

### CODE REVIEW

File: flame/train.py [hunk 1]
```diff
@@ -661,10 +659,7 @@ def main(job_config: JobConfig):
                             position_ids=position_ids,
                             cu_seqlens=cu_seqlens,
                         )
-                        loss = (
-                            output.loss
-                            / job_config.training.gradient_accumulation_steps
```
Comment (yzhangcs) [inline flame/train.py:666]: @kashif We should not remove / grad_accum_steps here IMO, as they're mean reduced, we should divide the subbatch loss by gradient_accumulation_steps to match the global batch one
